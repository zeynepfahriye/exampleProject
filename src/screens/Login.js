import { StyleSheet, SafeAreaView,View } from 'react-native'
import React from 'react'
import { Button, Card, TextInput,Text } from 'react-native-paper'
import { Formik } from 'formik'
import { loginForm } from './LoginForm'

const Login = ({navigation}) => {
const login = ()=> {navigation.navigate("Home")}
const register = ()=> {navigation.navigate("Register")}
  return (
   <SafeAreaView style={styles.content}>
   <View style ={styles.view}>
   <Card>
        <Card.Title title="Example app" titleStyle={styles.cardTitle}></Card.Title>
        <Card.Content>
        <Formik
        initialValues={{email:"",password:""}}
        onSubmit={login}
        validationSchema={loginForm}
        >
        {({handleSubmit,handleChange,errors,setFieldTouched,touched,values})=>(
            <>
            <TextInput
            style={{backgroundColor:'transparent'}} 
            label="Email" 
            keyboardType='email-address' 
            onChangeText={handleChange('email')}
            testID='email'
            onFocus={()=>setFieldTouched('email')}
            />
            {touched.email && errors.email ? <Text testID='error-email' style={{color:'red'}}>{errors.email}</Text>
            :null
            }
            <TextInput
             label="Password"
             style={{backgroundColor:'transparent'}}  
             secureTextEntry={true}
             onChangeText={handleChange('password')}
             testID='password'
             onFocus={()=>setFieldTouched('password')}
              />
              {touched.password && errors.password ? <Text testID='error-password' style={{color:'red'}}>{errors.password}</Text>
            :null
            }
            <Button testID='recoveryButton' uppercase={false} style={styles.cardButton} disabled={values.email =='' || errors.email ? true : false}>Forget email/password</Button>
            <Button testID='loginButton' onPress={handleSubmit} mode="contained" style={[styles.cardButton,{backgroundColor:'orange'}]}>Login</Button>
            <Button testID='registerButton' onPress={register} style={styles.cardButton}>Register</Button>
            </>
        )}
       
       </Formik>
        </Card.Content>
    </Card>
   </View>
   </SafeAreaView>
  )
}
const styles = StyleSheet.create({
    content :{
        display:'flex',
        flex:1,
        justifyContent:'center',
        alignItems:'center',
        flexDirection:'row',
        backgroundColor:'#e0b0ff'
    },
    view:{
        width:'85%'
    },
    cardTitle:{
        color:'purple'
    },
    cardButton:{
        margin:2,
        marginLeft:0,
        marginRight:0,
    }
})
export default Login