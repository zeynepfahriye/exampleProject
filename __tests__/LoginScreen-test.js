import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import Login from '../src/screens/Login'

test('Form header snapshot',()=>{
    const snap = renderer.create(
        <Login />
    ).toJSON();
    expect(snap).toMatchSnapshot();
})