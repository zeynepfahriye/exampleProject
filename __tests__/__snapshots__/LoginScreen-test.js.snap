// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form header snapshot 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e0b0ff",
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "width": "85%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "rgba(0, 0, 0, 0.12)",
          "borderRadius": 4,
          "elevation": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.5,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 0.75,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 16,
              },
              Object {
                "minHeight": 50,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 20,
                        "letterSpacing": 0.15,
                        "lineHeight": 30,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "minHeight": 30,
                          "paddingRight": 16,
                        },
                        Object {
                          "marginBottom": 2,
                        },
                        Object {
                          "color": "purple",
                        },
                      ],
                    ],
                  ],
                ]
              }
            >
              Example app
            </Text>
          </View>
          <View />
        </View>
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "paddingBottom": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(231, 231, 231)",
                  "borderTopLeftRadius": 4,
                  "borderTopRightRadius": 4,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.26)",
                  "bottom": 0,
                  "height": 2,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "transform": Array [
                    Object {
                      "scaleY": 0.5,
                    },
                  ],
                  "zIndex": 1,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "minHeight": 64,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "zIndex": 3,
                  }
                }
              >
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 34,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Email
                </Text>
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 34,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Email
                </Text>
              </View>
              <TextInput
                editable={true}
                keyboardType="email-address"
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=" "
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                selectionColor="#6200ee"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                    },
                    Object {
                      "paddingLeft": 12,
                      "paddingRight": 12,
                    },
                    Object {
                      "height": 64,
                    },
                    Object {
                      "paddingBottom": 4,
                      "paddingTop": 24,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontVariant": undefined,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="email"
                underlineColorAndroid="transparent"
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(231, 231, 231)",
                  "borderTopLeftRadius": 4,
                  "borderTopRightRadius": 4,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.26)",
                  "bottom": 0,
                  "height": 2,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "transform": Array [
                    Object {
                      "scaleY": 0.5,
                    },
                  ],
                  "zIndex": 1,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "minHeight": 64,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "zIndex": 3,
                  }
                }
              >
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 34,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Password
                </Text>
                <Text
                  collapsable={false}
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 34,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Password
                </Text>
              </View>
              <TextInput
                editable={true}
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=" "
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                secureTextEntry={true}
                selectionColor="#6200ee"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                    },
                    Object {
                      "paddingLeft": 12,
                      "paddingRight": 12,
                    },
                    Object {
                      "height": 64,
                    },
                    Object {
                      "paddingBottom": 4,
                      "paddingTop": 24,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontVariant": undefined,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="password"
                underlineColorAndroid="transparent"
              />
            </View>
          </View>
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 0,
                "margin": 2,
                "marginLeft": 0,
                "marginRight": 0,
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
              testID="recoveryButton"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        false,
                        Object {
                          "color": "rgba(0, 0, 0, 0.32)",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Forget email/password
                </Text>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "orange",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 2,
                "margin": 2,
                "marginLeft": 0,
                "marginRight": 0,
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0.75,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 1.5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
              testID="loginButton"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Login
                </Text>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 0,
                "margin": 2,
                "marginLeft": 0,
                "marginRight": 0,
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
              testID="registerButton"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#6200ee",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Register
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
